salt/modules/boto_cloudfront.py

```diff
@@ -0,0 +1,441 @@
+                {'tags': tags},
+            ]
+            results['Manage CloudFront distribution {0}'.format(name)] = {
+                'boto_cloudfront.present': distribution_sls_data,
+            }
+    except botocore.exceptions.ClientError as e:
+        # Raise an exception, as this is meant to be user-invoked at the CLI
+        # as opposed to being called from execution or state modules
+        raise e
+
+    dumper = __utils__['yamldumper.get_dumper']('IndentedSafeOrderedDumper')
+    return yaml.dump(
+        results,
+        default_flow_style=False,
+        Dumper=dumper,
+    )
+
+
+def create_distribution(
+    name,
+    config,
+    tags=None,
+    region=None,
+    key=None,
+    keyid=None,
+    profile=None,
+):
+    '''
+    Create a CloudFront distribution with the given name, config, and (optionally) tags.
+
+    name
+        Name for the CloudFront distribution
+
+    config
+        Configuration for the distribution
+
+    tags
+        Tags to associate with the distribution
+
+    region
+        Region to connect to
+
+    key
+        Secret key to use
+
+    keyid
+        Access key to use
+
+    profile
+        A dict with region, key, and keyid,
+        or a pillar key (string) that contains such a dict.
+    '''
+    if tags is None:
+        tags = {}
+    if 'Name' in tags:
+        # Be lenient and silently accept if names match, else error
+        if tags['Name'] != name:
+            return {'error': 'Must not pass `Name` in `tags` but as `name`'}
+    tags['Name'] = name
+    tags = {
+        'Items': [{'Key': k, 'Value': v} for k, v in six.iteritems(tags)]
+    }
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    try:
+        conn.create_distribution_with_tags(
+            DistributionConfigWithTags={
+                'DistributionConfig': config,
+                'Tags': tags,
+            },
+        )
+        _cache_id(
+            'cloudfront',
+            sub_resource=name,
+            invalidate=True,
+            region=region,
+            key=key,
+            keyid=keyid,
+            profile=profile,
+        )
+    except botocore.exceptions.ClientError as e:
+        return {'error': __utils__['boto3.get_error'](e)}
+
+    return {'result': True}
+
+
+def update_distribution(
+    name,
+    config,
+    tags=None,
+    region=None,
+    key=None,
+    keyid=None,
+    profile=None,
+):
+    '''
+    Update the config (and optionally tags) for the CloudFront distribution with the given name.
+
+    name
+        Name of the CloudFront distribution
+
+    config
+        Configuration for the distribution
+
+    tags
+        Tags to associate with the distribution
+
+    region
+        Region to connect to
+
+    key
+        Secret key to use
+
+    keyid
+        Access key to use
+
+    profile
+        A dict with region, key, and keyid,
+        or a pillar key (string) that contains such a dict.
+    '''
+    r = get_distribution(
```

Comment:
Would you mind changing this variable to `ret` or something along those lines instead of `r`? We like three-letter variables, according to our [style guide](https://docs.saltstack.com/en/latest/topics/development/conventions/style.html#variables).

I'll just comment in this function about this change, but there are a couple of other functions that need this update as well.